x/ecocredit/basket/codec.go

```diff
@@ -0,0 +1,17 @@
+package basket
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+)
+
+func RegisterTypes(registry codectypes.InterfaceRegistry) {
+	msgservice.RegisterMsgServiceDesc(registry, &_Msg_serviceDesc)
+}
+
+func RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&MsgCreate{}, "regen.basket/MsgCreate", nil)
+	cdc.RegisterConcrete(&MsgPut{}, "regen.basket/MsgPut", nil)
+	cdc.RegisterConcrete(&MsgTake{}, "regen.basket/MsgTake", nil)
```

Comment:
looks like we need to update `Type` function

Reply:
Which Type function?

Reply:
You mean Msg.Type? I'm pretty sure that's no longer used by the SDK but sure

Reply:
Yeah `Msg.Type` and `Msg.Route` can just be ignored. The SDK doesn't use them. We can just as well panic